src/line-graph.js

```diff
@@ -169,7 +174,54 @@ export class LineGraph extends HTMLElement {
       .attr("y", 15)
       .attr("transform", "rotate(-90)")
       .attr("fill", "#000");
+    this.setLegend(svg, height, data)
+    
+  }
+
+  /**
+   * 
+   * @param {d3 svg element} svg 
+   * @param {height} height 
+   * @param {data} data 
+   */
+  setLegend(svg, height, data) {
+    var legend = svg.append("g")
+      .attr("class", "legend")
+      .attr('transform', `translate(0,${height-50})`)
+    
+      // create rectangle for legends
+    legend.selectAll('rect')
+      .data(data)
+      .enter()
+      .append("rect")
+      .attr("x", 18)
+      .attr("y", (d, i) => {
+        return i * 20 + 30;
+      })
+      .attr("width", 10)
+      .attr("height", 10)
+      .style("fill", (d) => {
+        return d.color;
+      })
+      // set text of legends
+    legend.selectAll('text')
+      .data(data)
+      .enter()
+      .append("text")
+      .style("font-size",12)
+      .attr("x", 36)
+      .attr("y", (d, i) => {
+        return i * 20 + 38;
```

Comment:
what exactly is this line doing? lol

Reply:
Placing the legend text, if there are multiple lines then each line text must have some y-axis gap.
Do you any suggestions or alternate ways to handle it?

Reply:
No not at all. I just couldn't figure out what was being calculated. It looks good, thanks.